api/handler/v1beta1/job_run.go

```diff
@@ -140,14 +141,74 @@ func (sv *JobRunServiceServer) RegisterInstance(ctx context.Context, req *pb.Reg
 		Job:       jobProto,
 		Instance:  instanceProto,
 		Namespace: ToNamespaceProto(namespaceSpec),
-		Context: &pb.InstanceContext{
+		Context: &pb.JobRunInputResponse{
 			Envs:    jobRunInput.ConfigMap,
 			Secrets: jobRunInput.SecretsMap,
 			Files:   jobRunInput.FileMap,
 		},
 	}, nil
 }
 
+// JobRunInput is used to fetch task/hook compiled configuration and assets.
+func (sv *JobRunServiceServer) JobRunInput(ctx context.Context, req *pb.JobRunInputRequest) (*pb.JobRunInputResponse, error) {
+	projSpec, err := sv.projectService.Get(ctx, req.GetProjectName())
+	if err != nil {
+		return nil, mapToGRPCErr(sv.l, err, "not able to find project")
+	}
+
+	jobSpec, namespaceSpec, err := sv.jobSvc.GetByNameForProject(ctx, req.GetJobName(), projSpec)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s: job %s not found", err.Error(), req.GetJobName())
+	}
+
+	instanceType, err := models.ToInstanceType(utils.FromEnumProto(req.InstanceType.String(), "TYPE"))
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%s: instance type %s not found", err.Error(), req.InstanceType.String())
+	}
+
+	instanceName := req.GetInstanceName()
+
+	scheduledAt := req.GetScheduledAt().AsTime()
+
+	var jobRunSpec models.JobRunSpec
+	if req.JobrunId == "" {
+		jobRunSpec, err = sv.monitoringService.GetJobRunByScheduledAt(ctx, namespaceSpec, jobSpec, scheduledAt)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "%s: failed to get JobRun by ScheduledAt for job %s", err.Error(), jobSpec.Name)
+		}
+	} else {
+		jobRunID, err := uuid.Parse(req.JobrunId)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "%s: failed to parse uuid of job %s", err.Error(), req.JobrunId)
+		}
+		jobRunSpec, err = sv.monitoringService.GetJobRunByRunID(ctx, jobRunID)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "%s: failed to get JobRun by jobRunId::%s for job %s ", err.Error(), req.JobrunId, jobSpec.Name)
+		}
+	}
+	secrets, err := sv.secretService.GetSecrets(ctx, namespaceSpec)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "%s: failed to get secrets %s:%s", err.Error(), jobSpec.Name, namespaceSpec.Name)
+	}
+
+	jobRunInput, err := sv.jobRunInputCompiler.CompileNewJobSpec(ctx,
```

Comment:
want to have a small discussion to understand what are the inputs for compilation

Reply:
sure